### ISSUE TITLE
question of downsampling/sizing img datafiles before converting into num for numpy array necessary or not

### ISSUE THREAD

**Issue by nathiedeheyl** [nathiedeheyl/pp5_mildew-detection#16]
____________
Note to self
____________

First: import library and module to load, resize, and convert images into numbers (num arrays). This is needed for the computer to understand the picture(s).

**Important**: How to find out what size to resize the images to?

Loop through each label of each dataset, load and resize the images, then convert them to array and normalize them (which means to scale values of an array to a specific range). 

Finally, store image data and labels to x and y.
____________

This step prepares the image data for the training of the ml model.

The images are converted into numbers, that the model is able to process. Those images have to be the same size before training.

...

What Image Size to resize to? Determine the best image size.

```
from tensorflow.keras.preprocessing import image

img_path = "inputs/mildew_dataset/cherry-leaves/test/powdery_mildew/1a64abff-82d4-45dc-b984-f1a5e6a5229e___FREC_Pwd.M 4952_flipLR.JPG"
img = image.load_img(img_path)

img_resized_128 = img.resize((128, 128))
img_resized_64 = img.resize((64, 64))

plt.figure(figsize=(10, 5))
plt.subplot(1, 3, 1)
plt.imshow(img)
plt.title("Original")

plt.subplot(1, 3, 2)
plt.imshow(img_resized_128)
plt.title("Resized to 128x128")

plt.subplot(1, 3, 3)
plt.imshow(img_resized_64)
plt.title("Resized to 64x64")

plt.show()
```

...  I ran this in the jupyter notebook to have a look at the image files in different dimensions, before that with a healthy leaf, then with an infected one. 

... I came to the conclusion that 128 px should be fine to detect relevant features. 

I then tried to figure out how to downsize and load images into a numpy array as the walkthrough does, when it dawned on me that the walkthrough does not seem to downsize the images...

**Comment by nathiedeheyl** [comment]
![Image](https://github.com/user-attachments/assets/49d103bc-fdf1-4024-b341-dd29fa20d803)

**Comment by nathiedeheyl** [comment]
### What I tried to solve the issue (1)

`image.load_img(img_path, target_size=new_size)` is where the img data should be resized. Added print statement `print(f"Loaded image size: {img.size}")` of which the output still is => "Loaded image size: (256, 256)"

The tensorflow documentation lists "target_size" as an argument that takes "tuple of ints (img_height, img_width)", so I tried replacing the var "new_size" by its values directly to see if resizing would work then, with this result: 

![Image](https://github.com/user-attachments/assets/0209bae9-274e-41bc-a66e-869155a9f39c)

Which suggest that the resizing works now, but the resized result cannot be converted into a numpy array any longer.

**Comment by nathiedeheyl** [comment]
### How I solved the issue

Eventually, I ended up understanding that while calling the function, the argument new_size then passes the value of variable image_shape that has been defined earlier and that then did actually solve it ... 

![Image](https://github.com/user-attachments/assets/596e1aef-16f3-4c91-8460-16ead8a60a75)
